### PR TITLE
Fix proxy protocol header

### PR DESCRIPTION
**Pull request by AdamEr8**
TL;DR: HAProxy's PROXY protocol header was format had a minor mix-up, this PR should fix it.

I tried using this tcpproxy library on a golang project that required using HAProxy's PROXY protocol,
but something didn't work for me.

looking through the code and PROXY protocol documentation it seemed that:
original code sent the header in the format: PROXY family srcIP srcPort dstIP dstPort
according to docs header format should be: PROXY family srcIP dstIP srcPort dstPort

this is according to:
https://www.haproxy.org/download/1.8/doc/proxy-protocol.txt
section 2.1. Human-readable header format (Version 1).

after modifying the code, we successfully used PROXY protocol with our HAProxy server.

### CODE REVIEW

**Comment by AdamEr8** [comment]
Travis checks failed, both sub-jobs (90.1 and 90.2) have the error:
"The command "go get github.com/golang/lint/golint" failed [....]"

which I suppose is not related to my changes. I'd love to get some help from someone to fix this issue 😄 


Job 90.1: https://travis-ci.org/google/tcpproxy/jobs/493623519
Job 90.2: https://travis-ci.org/google/tcpproxy/jobs/493623520

**Comment by bradfitz** [comment]
@danderson, we should not have `lint` in our Travis presubmit checks. (That was Rob Pike's decree on the linter from day 1 of the lint tool existing, actually.) There are four levels of checkers:

* the compiler catches language violations, and also things the spec says are okay for compiler implementations to fail on (e.g. assigned but never read variables)
* `vet`'s always-correct no-false-positive checks (as enabled by default by `go test` a few releases ago)
* `vet`'s probably-correct checks with some false positives requiring human review
* and finally, `lint`, which is full of noise and always requires human review

We definitely shouldn't have `lint` breaking a build.

**Comment by danderson** [comment]
Agree. Also our CI is deeply bitrotted, it's still trying to build with Go 1.8. I'll fix the CI now. @AdamEr8 when I've pushed a CI fix, can you rebase and repush so we can get a happy CI signal?

**Comment by AdamEr8** [comment]
> @AdamEr8 when I've pushed a CI fix, can you rebase and repush so we can get a happy CI signal?

Thanks! will do 👍

**Comment by discostur** [comment]
can you please merge this fix? proxy protocol header is still broken ...

**Comment by AdamEr8** [comment]
> Agree. Also our CI is deeply bitrotted, it's still trying to build with Go 1.8. I'll fix the CI now. @AdamEr8 when I've pushed a CI fix, can you rebase and repush so we can get a happy CI signal?

Hi @danderson, I know this is probably not a very high priority, but any chance the CI is already fixed?

Thanks,

**Comment by AdamEr8** [comment]
trying to close and reopen to try and trigger another Travis CI build, to see if anything changed

**Comment by AdamEr8** [comment]
trying to close and reopen to try and trigger another Travis CI build, to see if anything changed

**Comment by danderson** [comment]
It hasn't, but I haven't had time to give it care and feeding. But the change looks reasonable, so let's just go with it.

**Comment by danderson** [comment]
... and I no longer have repo access? @bradfitz fix plz?

**Comment by AdamEr8** [comment]
closing and re-opening to bump again :)

**Review by bradfitz — APPROVED** [review]

**Comment by dnwe** [comment]
@danderson I think this PR could be merged now. Ran into this issue myself when passing through to nginx with PROXY protocol enabled. After applying this patch things started working.

**Comment by danderson** [comment]
Done, thanks!
